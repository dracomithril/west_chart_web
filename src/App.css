.App {
    text-align: center;
    min-width: 760px;
    margin-left: 1rem;
    margin-right: 1rem;
}

.App-body {
    padding: 5px 5px;
    margin: 0 10px 50px;
    /*background-color: #eee;*/
    /*float: left;*/
    /*width: 80%;*/
}

.btn {
    /*margin-bottom: 10px;*/
    /*margin-top: 5px;*/
}
.console-log{
    border: 5px dashed red;
    background-color: indianred;
    margin: 10px;
    padding: 0 5px 5px 5px;
}
.checkbox1 {
    position: relative;
    display: block;
    /*margin-top: 10px;*/
    /*margin-bottom: 10px;*/
}

.checkbox1 input[type=checkbox]:checked {
    background-color: green;
}

#updateDate {
    color: red;
    margin-left: 10px;
}

.formArea {
    width: 240px;
    margin-bottom: 5px;
    margin-left: 5px;
    float: left;
}

.label {
    margin: 0;
    padding: .6em .9em .5em;
}

.likes {
    display: inline-block;
    min-width: 10px;
    padding: 3px 7px;
    font-size: 12px;
    font-weight: 700;
    line-height: 1;
    color: #fff;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    background-color: #337ab7;
    border-radius: 10px;
}

.elements_count {
    display: inline-block;
    min-width: 10px;
    padding: 3px 7px;
    font-size: 12px;
    font-weight: 700;
    line-height: 1;
    color: #fff;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    background-color: #b7001e;
    border-radius: 10px;
}

.popover {
    max-width: 600px;
}
.datePiker input[type="text"]{
    padding: 0;
}
.datePiker input[type="text"]:disabled {
    background-color: rgb(235, 235, 228);
}

.react-datepicker__input-container input {
    width: 80px;
}

#d_until {
    margin-left: 50px;
}

.spotify_sumary {
    border: 2px solid blue;
    border-radius: 10px;
    padding: 4px;
    margin: 4px;
}

.spotify_sumary span {
    color: darkgreen;
    font-size: medium;
}

.modal {
    top: 300px
}

