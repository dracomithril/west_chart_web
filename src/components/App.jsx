import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import CookieBanner from 'react-cookie-banner';

import Footer from './Footer';
import Navigation from './Navigation';
import createStore from '../configureStore';

const store = createStore();

class App extends React.Component {
  componentDidMount() {
    fetch('api/info')
      .then(resp => resp.text())
      .then((resp) => {
        console.info('response from api:', resp);
      })
      .catch((err) => {
        console.error('failed to call api :(', err.message);
      });
  }

  render() {
    return (
      <Provider store={store}>
        <Router>
          <div className="App">
            <CookieBanner
              message={"Yes, we use cookies. If you don't like it change website, we won't miss you! ;)"}
              cookie="user-has-accepted-cookies"
            />
            <Navigation />
            <Footer />
          </div>
        </Router>
      </Provider>
    );
  }
}

App.contextTypes = {
  store: PropTypes.shape(),
};
export default App;
