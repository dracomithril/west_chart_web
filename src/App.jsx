import React from 'react';
import PropTypes from 'prop-types';
import { Nav, NavItem } from 'react-bootstrap';
import 'react-table/react-table.css';
import 'react-datepicker/dist/react-datepicker.css';
import { Redirect, Route } from 'react-router-dom';
import ChartPresenter from './components/ChartPresenter';
import './App.css';
import ErrorConsole from './components/ErrorConsole';
import Header from './components/Header';
import Footer from './components/Footer';
import Combiner from './components/PlaylistCombiner';
import LoginAlert from './components/LoginAlert';

const pathways = ['/', '/chart', '/combiner'];

const PrivateRoute = ({ component: Component, ...rest }, { store }) => {
  const { user, sp_user } = store.getState();
  const isAuthenticated = !!user.id && !!sp_user.id;
  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: '/login',
              state: { from: rest.path },
            }}
          />
        )
      }
    />
  );
};
PrivateRoute.propTypes = {
  component: PropTypes.func,
  location: PropTypes.object,
};
PrivateRoute.contextTypes = {
  store: PropTypes.object,
};

const About = () => (
  <div>
    <h2>Hi That will be introduction</h2>
    <h3 style={{ color: 'red' }}>Creation in progress</h3>
    <h4 style={{ color: 'gray' }}>Nothing is true everything is permitted</h4>
  </div>
);

const Navigation = () => (
  <div>
    <Nav bsStyle="tabs" activeKey={(() => pathways.indexOf(window.location.pathname))()}>
      <NavItem eventKey={0} href="/">
        Info
      </NavItem>

      <NavItem eventKey={1} href="/chart">
        Chart
      </NavItem>
      <NavItem eventKey={2} href="/combiner">
        Combiner(BETA)
      </NavItem>
    </Nav>
    <Route exact path="/" component={About} />
    <Route path="/login" component={LoginAlert} />
    <PrivateRoute path="/chart" exact component={ChartPresenter} />
    <PrivateRoute path="/combiner" exact component={Combiner} />
  </div>
);

class App extends React.Component {
  componentDidMount() {
    const { store } = this.context;
    this.unsubscribe = store.subscribe(() => this.forceUpdate());
  }
  componentWillUnmount() {
    this.unsubscribe();
  }

  render() {
    return (
      <div className="App">
        <Header />
        <ErrorConsole />
        <Navigation />
        <Footer />
      </div>
    );
  }
}

App.contextTypes = {
  store: PropTypes.object,
};
export default App;
