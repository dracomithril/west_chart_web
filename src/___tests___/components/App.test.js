import React from 'react';
import PropTypes from 'prop-types';
import App from './../../App';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

Enzyme.configure({ adapter: new Adapter() });
const initial_state = require('../data/initial_state.json');

const middlewares = [];
const mockStore = configureMockStore(middlewares);
describe('<App/>', () => {
  it('renders without crashing', () => {
    const store = mockStore(initial_state);
    const wrapper = shallow(
      <App/>, {
        context: { store },
        childContextTypes: { store: PropTypes.object }
      }
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
