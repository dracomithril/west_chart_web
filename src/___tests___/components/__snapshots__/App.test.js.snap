// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div
      className="App"
    >
      <Header />
      <Navigation />
      <Footer />
    </div>
  </BrowserRouter>
</Provider>
`;
